src/main/scala/com/databricks/labs/overwatch/utils/SchemaTools.scala

```diff
@@ -123,10 +123,7 @@ object SchemaTools extends SparkSessionWrapper {
       println(s"WARNING: The json schema for column $c was not parsed correctly, please review.")
     }
     if (jsonSchema.isEmpty) {
-      if (minSchema.isEmpty) logger.log(Level.WARN, s"A schema cannot be inferred for $c give the data " +
-        s"in this dataframe. A minimum schema may be provided but hasn't been. Will implicitly cast $c to a null of " +
-        s"StringType. If errors persist, please review this field.")
-      lit(null).cast(minSchema.getOrElse(StringType))
+      lit(null)
```

Comment:
that's dangerous - Delta just silently drop null columns, so when you read data back you may not have it. maybe use `lit(null).cast("struct<>")` if we expect to return a struct column?

Reply:
That is exactly the behavior I want. If the dynamic schema does have data for the column, drop it. 
Example:  Col(x) has a string value of "{}" (empty json) there is no data thus the source dataframe will not include that column at all. If it's a required column downstream, the `df.verifyMinimumSchema(schema)` will add it back and type it correctly.